new_samples/main.go

```diff
@@ -0,0 +1,18 @@
+package main
+
+import (
+	"fmt"
+	"github.com/uber-common/cadence-samples/new_samples/worker"
```

Comment:
As a more general thought: is there any reason we are _preferring_ multiple packages instead of putting it all in one?

I can kinda see arguments in either direction (e.g. most real-world things will have multiple packages), and regardless this is all FAR simpler than the previous patterns.  Just curious if there was a reason.

Reply:
I think it is more likely to be a way that how same functionalities are grouped together. I'm ok to load everything into a main but I feel like the main should be just the entry point of an application. Registering workers and implementing works are two different topics so that's why I split them. Another reason is that I would like to make the code slightly easier to be searched. As I put into the workflow package, users can just come to this repo and search the workflows folder. Maybe this will make their life easier?

Reply:
yeah, separate _files_ I think makes complete sense.  main / worker / workflow are completely reasonable splits.  that doesn't need different _packages_ though.

but given the other thread, sounds like the plan is to put everything into one binary.  in that case, yeah, packages give it a bit more structure and that seems fine 👍